### ISSUE TITLE
Week-45 - Consolidation of WLNM Model Variants & Evaluation Regimes

### ISSUE THREAD

**Issue by EduardoCastro15** [EduardoCastro15/qmul-phd-framework#34]
## **Summary**

Streamline the WLNM experimental framework by collapsing the current eight variants into **four core models**, while reframing the remaining variants as **evaluation regimes** or **protocols**.
The goal is to produce a clear, defensible modeling taxonomy that highlights the key conceptual contributions—**directionality** and **ecology-aware negatives**—and to standardize evaluation through rare-link and backbone regimes.
This update also formalizes a new **“evaluate-on-all-unseen” protocol**, which expands the test set to include all unknown links for exhaustive evaluation of generalization ability.
Together, these changes prepare the framework for final large-scale reruns and figure generation for the manuscript.

---

## **Acceptance Criteria**

* [ ] Refactor configuration and execution logic to support **four canonical WLNM model variants**:

  1. **WLNM – Original** (baseline)
  2. **WLNM – Directed** (preserves edge directionality)
  3. **WLNM – Negative** (ecology-aware negative sampling)
  4. **WLNM – Directed + Negative** (flagship model)

* [ ] Apply the following as **protocols** rather than standalone models:

  * **Training-ratio sweep** → systematic protocol (10%→90%) applied to all four models.
  * **Small-network exclusion** → robustness sensitivity filter applied post-hoc.
  * **Evaluate-on-all-unseen** → exhaustive protocol evaluating test negatives over *all unseen links* (activated via `evaluate_on_all_unseen = true` in `sample_neg_*`).

* [ ] Reframe and document **evaluation regimes**:

  * **Rare-links (master-web)** → applied primarily to *Directed + Negative*, baseline for delta.
  * **Backbone-conditioned** → applied primarily to *Directed + Negative*, baseline for reference.

* [ ] Generate a concise **results matrix** summarizing:

  * 4 models × 2 regimes × 3 protocols × 5 core metrics (AUC, Precision, Recall, F1, PR-AUC).
  * Aggregated per ecosystem type and as overall mean ± std.

* [ ] Produce summary visualizations:

  * **Main text:** comparative plots across the 4 models (AUC / F1 per ecosystem).
  * **Supplementary:** regime-specific (rare-link, backbone) and protocol sweeps (training-ratio, small-network, all-unseen).
  * Save as `.svg` and `.png`.

* [ ] Update **documentation and README**:

  * Add “Model Taxonomy” section explaining the consolidation rationale.
  * Add “Evaluation Regimes” section describing rare-link and backbone setups.
  * Add “Protocols” section (training-ratio, small-network, evaluate-on-all-unseen) under reproducibility notes.

---

## **Deliverables**

* Results CSVs in `results/week-45/` structured as:

  ```
  results/week-45/
    ├── model_directed_negative/
    │    ├── regime_backbone/
    │    ├── regime_rare_links/
    │    └── protocols/
    │         ├── ratio_sweep/
    │         ├── small_network/
    │         └── all_unseen/
  ```
* Summary plots:

  * `plots/week-45/model_comparison_AUC.svg`
  * `plots/week-45/regime_backbone_vs_rarelinks.svg`
  * `plots/week-45/protocol_all_unseen_vs_standard.svg`

---

## **Notes / Rationale**

* The variant list is reduced from **8 → 4**, simplifying interpretation while retaining conceptual diversity.
* The **evaluation regimes** (rare-links, backbone) target ecological realism.
* The **protocols** ensure methodological robustness and reproducibility:

  * *Training-ratio sweep* – sensitivity to training data size.
  * *Small-network exclusion* – robustness to undersized webs.
  * *Evaluate-on-all-unseen* – stress test of model generalization when every unseen link is treated as negative.
